admin/services/class-indexable.php

```diff
@@ -19,43 +19,51 @@ class WPSEO_Indexable_Service {
 	 */
 	public function get_indexable( WP_REST_Request $request ) {
 		$object_type = $request->get_param( 'object_type' );
-		$provider    = $this->get_provider( strtolower( $object_type ) );
+		$object_id 	 = $request->get_param( 'object_id' );
 
-		if ( $provider === null ) {
-			return new WP_REST_Response(
-				sprintf(
-					/* translators: %1$s expands to the requested indexable type  */
-					__( 'Unknown type %1$s', 'wordpress-seo' ),
-					$object_type
-				),
-				400
-			);
-		}
+		try {
+			$provider  = $this->get_provider( $object_type );
+			$indexable = $provider->get( $object_id );
 
-		$object_id = $request->get_param( 'object_id' );
-		if ( ! $provider->is_indexable( $object_id ) ) {
-			return new WP_REST_Response(
-				sprintf(
-					/* translators: %1$s expands to the requested indexable type. %2$s expands to the request id */
-					__( 'Object %1$s with id %2$s not found', 'wordpress-seo' ),
-					$object_type,
-					$object_id
-				),
-				404
-			);
+			return new WP_REST_Response( $indexable );
+		} catch ( Exception $exception ) {
+			return new WP_REST_Response( $exception->getMessage(), 500 );
 		}
+	}
+
+	/**
+	 * Patches an indexable with the request parameters.
+	 *
+	 * @param WP_REST_Request $request The REST API request to process.
+	 *
+	 * @return WP_REST_Response The REST response.
+	 */
+	public function patch_indexable( WP_REST_Request $request ) {
+		$object_type = $request->get_param( 'object_type' );
+		$object_id 	 = $request->get_param( 'object_id' );
 
-		return new WP_REST_Response( $provider->get( $object_id ) );
+		try {
+			$provider = $this->get_provider( $object_type );
+			$provider->patch( $object_id, $request->get_params() );
+
+			return new WP_REST_Response( 'Patch successful' );
+		} catch ( Exception $exception ) {
+			return new WP_REST_Response( $exception->getMessage(), 500 );
+		}
 	}
 
 	/**
 	 * Returns a provider based on the given object type.
 	 *
 	 * @param string $object_type The object type to get the provider for.
 	 *
-	 * @return null|WPSEO_Indexable_Service_Provider Instance of the service provider.
+	 * @return WPSEO_Indexable_Service_Provider Instance of the service provider.
+	 *
+	 * @throws WPSEO_Invalid_Argument_Exception The invalid argument exception.
 	 */
 	protected function get_provider( $object_type ) {
+		$object_type = strtolower( $object_type );
+
 		if ( $object_type === 'post' ) {
 			return new WPSEO_Indexable_Service_Post_Provider();
```

Comment:
Object type providers should be filterable IMO to support adding more, e.g. for custom database tables.

Reply:
I am not sure if this is something we should support for now. It needs some extra validation. Having that said, if we implement well with an interface and use `instanceof` to verify it might be something we can consider. /cc @atimmer @herregroen @moorscode